docs/openapi/schemas.yaml

```diff
@@ -1517,3 +1520,91 @@ ImportAssistantResponse:
     choices:
       - message:
           content: '.breadcrumbs'
+ApiKeyRequest:
+  type: object
+  required:
+    - name
+    - feature
+    - domains
+  properties:
+    name:
+      description: The name of the API key
+      type: string
+    feature:
+      description: The feature the API key is for
+      type: string
+    domains:
+      description: The domains the API key is allowed to access
+      type: array
+      items:
+        type: string
+  example:
+    name: 'Example API Key'
+    feature: 'imports'
+    domains:
+      - 'example.com'
+ApiKeyResponse:
+  type: object
+  properties:
+    id:
+      description: The ID of the API key
+      $ref: '#/Id'
+    name:
+      description: The name of the API key
+      type: string
+    apiKey:
+      description: The API key
+      type: string
+    scopes:
+      description: The scopes of the API key
+      type: array
+      items:
+        type: string
+    domains:
+      description: The domains the API key is allowed to access
+      type: array
+      items:
+        type: string
+    imsOrgId:
+      description: The IMS organization ID
+      $ref: '#/ImsOrganizationId'
+    imsUserId:
+      description: The IMS user ID
+      $ref: '#/ImsUserId'
+    createdAt:
+      description: The date and time when the API key was created
+      $ref: '#/DateTime'
+    expiresAt:
+      description: The date and time when the API key expires
+      $ref: '#/DateTime'
+ApiKeyMetadataResponse:
```

Comment:
Does this type just hide the `apiKey` value? Any way we could accomplish this without the duplication?

Reply:
Something like this:
```
ApiKeyResponse:
  type: object
  properties:
    id:
      description: The ID of the API key
      $ref: '#/Id'
    name:
      description: The name of the API key
      type: string
    apiKey:
      description: The API key
      type: string
    scopes:
      description: The scopes of the API key
      type: array
      items:
        type: string
    domains:
      description: The domains the API key is allowed to access
      type: array
      items:
        type: string
    imsOrgId:
      description: The IMS organization ID
      $ref: '#/ImsOrganizationId'
    imsUserId:
      description: The IMS user ID
      $ref: '#/ImsUserId'
    createdAt:
      description: The date and time when the API key was created
      $ref: '#/DateTime'
    expiresAt:
      description: The date and time when the API key expires
      $ref: '#/DateTime'

ApiKeyMetadataResponse:
  allOf:
    - $ref: '#/ApiKeyResponse'
    - type: object
      properties:
        apiKey: {}
      additionalProperties: false
```

Reply:
But, in this case, it shows the apiKey as null.
![Screenshot 2024-10-24 at 10 44 26 AM](https://github.com/user-attachments/assets/a7902357-6b40-4943-a553-1395c64a3ec6)

Reply:
Done.